orkamv_visibility_toggle_controller.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+ ORKa.MV Data API
+                                 A QGIS plugin
+ This plugin downloads a geopackage from the ORKa.MV Data API
+ Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                              -------------------
+        begin                : 2022-01-19
+        git sha              : $Format:%H$
+        copyright            : (C) 2022 by terrestris GmbH & Co. KG
+        email                : info@terrestris.de
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+import os
+from typing import Optional, List, Dict
+
+from qgis.PyQt.QtCore import QSettings, QTranslator, QCoreApplication
+from PyQt5.QtWidgets import QCheckBox, QWidget, QVBoxLayout, QSizePolicy
+from PyQt5.QtCore import Qt
+from qgis.gui import QgisInterface
+from qgis.core import QgsProject, QgsLayerTreeGroup, QgsLayerTreeLayer
+
+from .types import LAYER_GROUPS_PROP_NAME, LayerGroup
+from .orkamv_visibility_toggle_plugin_dialog import OrkamvVisibilityTogglePluginDialog
+
+
+class OrkamvVisibilityToggleController:
+
+    iface: QgisInterface
+    dlg: Optional[OrkamvVisibilityTogglePluginDialog] = None
+
+    selected_layer_group: str = None
+    selected_groups_hash: Dict[str, List[str]] = {}
+
+    def __init__(self, iface: QgisInterface):
+        self.iface = iface
+        # initialize plugin directory
+        self.plugin_dir = os.path.dirname(__file__)
```

Comment:
Would be nice if all class members were annotated with types in the beginning of the class.

Reply:
solved in https://github.com/terrestris/orkamv-data-api-plugin/pull/13/commits/df45de6373a72276ad0f9d50054a2a28bb700fec

Reply:
Will be part of future refactoring